### DISCUSSION TITLE
verify command does not verify certificate chain

### DISCUSSION THREAD

**Discussion by Torkolis** [openssl/openssl#25045] [Q&A]
I have a certificate chain of 3 certificates: root -- intermediate -- server
How it is supposed to work is that I should be able to verify the server certificate with the root certificate as long as the server itself delivers all the intermediate certificates (in this case just 1) with the server certificate.
So what i do is that I have 2 files: ca.cert containing only the root certificate and server.cert containing the server certificate followed by the intermediate certificate. This is what works with all the default web servers and web clients e.g. when I start a https server with nginx using the server.cert (containing both certificates) and import **only** the CA into my trusted CA store of the OS -> Browsers trust the server.

If I use openssl however (with `openssl verify -CAfile ca.cert server.cert`), the verification fails with 
`error 20 at 0 depth lookup: unable to get local issuer certificate`
only if I add the intermediate certificate to the ca.cert file it works. This is not as designed and should be changed, openssl should rebuild the chain with certificates delivered by the client/cert-file-to-verify and verify it using **only** the root certificate.

**Comment by t8m** [comment]
Use `openssl verify -CAfile ca.cert -untrusted server.cert server.cert`. This should verify the chain as expected.

**Reply by Torkolis** [reply]
The problem is with dependent software, e.g. mariadb (client CLI) when establishing the connection, I think it even uses `openssl s_client` and not `openssl verify` because they assume openssl follows the rules, but it does not.
